modules/component-store/src/component-store.ts

```diff
@@ -308,6 +314,26 @@ export class ComponentStore<T extends object> implements OnDestroy {
       });
     }) as unknown as ReturnType;
   }
+
+  /**
+   * Used to check if lifecycle hooks are defined
+   * but not used with provideComponentStore()
+   */
+  private checkProviderForHooks() {
+    asyncScheduler.schedule(() => {
+      if (
+        isDevMode() &&
+        (isOnStoreInitDefined(this) || isOnStateInitDefined(this)) &&
+        !this.ɵhasProvider
+      ) {
+        console.warn(
+          `@ngrx/component-store: The ${this.constructor.name} has one or more ` +
+            'lifecycle hooks implemented without being provided using the ' +
+            `provideComponentStore(${this.constructor.name}) function. `
+        );
+      }
+    });
+  }
```

Comment:
To help the devs with this warning, I prefer to make it more explicit.
Feel free to ignore though.

```suggestion
    asyncScheduler.schedule(() => {
      if (
        isDevMode() &&
        (isOnStoreInitDefined(this) || isOnStateInitDefined(this)) &&
        !this.ɵhasProvider
      ) {
        const warnings = [
          isOnStoreInitDefined(this) ? 'OnStoreInit' : ''
          isOnStateInitDefined(this) ? 'OnStateInit' : '' 
        ].filter(Boolean);
        
        console.warn(
          `@ngrx/component-store: ${this.constructor.name} has the ${warnings.join(' and ')} ` +
            'lifecycle hook(s) implemented without being provided using the ' +
            `provideComponentStore(${this.constructor.name}) function. ` +
            `To resolve this, provide the component store via provideComponentStore(${this.constructor.name})`
        );
      }
    });
  }
```